presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotConnection.java

```diff
@@ -110,4 +110,19 @@ public TimeBoundary getTimeBoundary(String tableName)
     {
         return pinotClusterInfoFetcher.getTimeBoundaryForTable(tableName);
     }
+
+    public int getGrpcPort(String serverInstance)
+    {
+        int grpcPort;
+        try {
+            grpcPort = pinotClusterInfoFetcher.getGrpcPort(serverInstance);
+            if (grpcPort <= 0) {
+                grpcPort = pinotConfig.getServerGrpcPort();
+            }
+        }
+        catch (Exception e) {
+            grpcPort = pinotConfig.getServerGrpcPort();
```

Comment:
When will this exception be hit ? My understanding is that when the controller does not have the per instance grpc port, it will simply return -1 for that and we will go through the regular if condition on line 119. I don't know whether we should fall back in case of an exception. Are you thinking of cases like where the controller is down or such ?

Can we catch a more specific exception ?

Reply:
The only thing I can think of is that the pinot controller is unavailable so the http call will fail.

Reply:
I could be wrong, but I think that the rest of the code is also susceptible to failing when the controller fails. So it may be okay to just propagate the exception, since it is not like you are adding a new vulnerability.

Where is the Pinot Server's GRPC port specified in the Pinot configuration (ie on the Pinot cluster side): If there too, the port is fixed (how does that work for k8s then, which does not like fixed ports), then I can see why having a fall back port specified in the pinot config makes sense. 

I don't have a strong feeling here, and I am totally fine with having this config. It just seems weird to me to use some pre-configured port value when we couldn't get the real thing :-). Either way the query will fail.

Reply:
It's a fixed port that user can specify in pinot server config, then it will be stored in Zookeeper and pinot controller provides API to retrieve it. 
This is the zk record:
![image](https://user-images.githubusercontent.com/1202120/97091312-f2372780-15ef-11eb-8c0a-8726900ca547.png)

Reply:
> It's a fixed port that user can specify in pinot server config, then it will be stored in Zookeeper and pinot controller provides API to retrieve it.
> This is the zk record:
> ![image](https://user-images.githubusercontent.com/1202120/97091312-f2372780-15ef-11eb-8c0a-8726900ca547.png)

I rethink this one, if in case of controller is down, we don't need to handle the case, then it' ok to remove this config.